src/rust/receiver-mock/src/router.rs

```diff
@@ -68,6 +70,22 @@ impl AppState {
             }
         }
     }
+    pub fn add_log_lines<'a>(&self, lines: impl Iterator<Item = &'a str>, ipaddr: IpAddr, opts: &options::Options) {
+        {
+            let mut message_count = 0;
+            let mut byte_count = 0;
+            let mut log_messages = self.log_messages.write().unwrap();
+            for line in lines {
+                message_count += 1;
+                byte_count += line.len() as u64;
+                if opts.store_logs {
+                    log_messages.add_log_message(line.to_string())
+                }
+            }
+            let mut log_stats = self.log_stats.write().unwrap();
+            log_stats.update(message_count, byte_count, ipaddr);
+        }
```

Comment:
Why the extra braces?

```suggestion
        let mut message_count = 0;
        let mut byte_count = 0;
        let mut log_messages = self.log_messages.write().unwrap();
        for line in lines {
            message_count += 1;
            byte_count += line.len() as u64;
            if opts.store_logs {
                log_messages.add_log_message(line.to_string())
            }
        }
        let mut log_stats = self.log_stats.write().unwrap();
        log_stats.update(message_count, byte_count, ipaddr);
```

Reply:
Copypaste fail